website/content/040-api/01-nexus/01-schema.mdx

```diff
@@ -944,6 +948,30 @@ schema.queryType({
 })
 ```
 
+Forwarding the incoming request to your GraphQL Context
```

Comment:
Why don't we take the opportunity to do this now as well? That is just put req and res right on the ctx by default all the time.

Reply:
It's a bit tricky because of the context type extractor. A simple way could be to make the `req` and `res` "reserved keys" and forbid users to overwrite them completely. Cause if we allow them to do so, it's gonna be tricky to let the extracted types overwrite the default types (of the `req` and `res`), as we don't extract any metadata about the fields, we just print them. And if we don't do any validation, we might end up in this situation:

```ts
interface Context {
  req: Request
  res: Response
}

interface Context {
  req: number // Subsequent property declarations must have the same type.  Property 'req' must be of type 'Request', but here has type 'number'
}
```

Reply:
IIUC this is a problem we have right now anyways.

Does putting req/res on the context increase the risk in practice right now you think?

About the underlying problem, there's a whole little discussion to have there at some point.

Reply:
To be clear this is definitely a separate feature and we do not at all need to hold up this PR for it of course! Thought it might be a cheap addon but maybe not.

Reply:
> Does putting req/res on the context increase the risk in practice right now you think?

Maybe you're right and at worse they're just gonna duplicate the interface

Reply:
Since it doesn't feel obvious and we're a few comments, let's save this for a future PR. I'm not even sure about the use-case. Just a lot of users ask for it. But pretty sure none of the use-cases are for great reasons. But I'm thinking about it as "let's ship something" now rather than waiting more and more.